### PR TITLE
Feature/admin route

### PR DESCRIPTION
**Pull request by maxgrossman**
new route with endpoints that 
- filter admin areas by matchingn leading string
- list admin areas by level (commune, district, province)
- provide information (level, name, id, children and parent units, and bbox) for provided admin area

### CODE REVIEW

File: routes/admin.js [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+     *    ...
+     *  ]
+     * }
+    */
+    method: 'GET',
+    path: '/admin/{level}/units',
+    handler: function (req, res) {
+      const level = req.params.level;
+      // check to make sure admin level is of the proper type
+      const levelCheck = ['commune', 'district', 'province'].find((l) => { return l === level; });
```
Comment (dereklieu) [inline routes/admin.js]: You should probably just use `Array.indexOf` here, I think it's more common for what this is doing.

File: routes/admin.js [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+      if (!levelCheck) {
+        res(Boom.badRequest('Admin level must be of type \'province\', \'district\', or \'commune'));
```
Comment (dereklieu) [inline routes/admin.js]: Have you checked whether the rest of the code below will still run if this conditional is triggered?  I'm not sure if you have to `return res(Boom.badRequest)` or not.

File: routes/admin.js [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+        // return error if it occurs
+        res(Boom.wrap(e));
+      });
+    }
+  },
+  {
+    /**
+     * @api {get} /admin/units?name=substring&limit=# List Matching Units
+     * @apiGroup Admin
+     * @apiName ListMathcingUnits
```
Comment (dereklieu) [inline routes/admin.js]: Typo

File: routes/admin.js [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+     * @apiGroup Admin
+     * @apiName ListAdmins
+     * @apiDescription Returns list of admin unit ids of specified level.
+     * @apiVersion 0.1.0
+     *
+     * @apiParam {String} level Admin level (province, district, or commune)
+     *
+     * @apiSuccess {JSON} units object including list of all admin ids in the level supplied in parameters
+     * @apiSuccess {string} units.level admin units' level
+     * @apiSuccess {array} units.units list of objects representing all units in level. each object icludes an id and name
```
Comment (dereklieu) [inline routes/admin.js]: Isn't the response `{ [level]: [{ name_en, id }, { name_en, id } ...] }`?

File: routes/admin.js [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+      console.log(queryString);
+      const limit = req.query.limit;
+      // selects all admin boundaries ids and name_en fields where name_en field matches query string
+      knex('admin_boundaries')
+      .where('name_en', 'ILIKE', `${queryString}%`)
+      .select('id', 'name_en', 'type as level')
+      .then((results) => {
+        // serve the response. if a limit is found in the query string, limit response to that limit
+        // if not, just serve all that was found
+        limit ? res(results.slice(0, Number(limit))) : res(results);
```
Comment (dereklieu) [inline routes/admin.js]: It'd be good practice to set a default limit that is also the maximum allowed and pass it in the knex query using `.limit(limit)`.

Comment (dereklieu) [reply routes/admin.js]: Also you might want to check explicitly whether `limit` can be coerced to a number using `!isNaN(limit)`

File: routes/admin.js [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+     * }
+     *
+    */
+    method: 'GET',
+    path: '/admin/{unit_id}/info',
+    handler: function (req, res) {
+      // get unitId from request params
+      const unitId = req.params.unit_id;
+      // check if unitId is valid length
+      const validIdCheck = [7, 5, 3].find(idLenght => idLenght === unitId.toString().length);
```
Comment (dereklieu) [inline routes/admin.js]: Same thing about `indexOf` here.

File: routes/admin.js [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+        let reposnse = {
+          id: info[0].id,
+          name: info[0].name,
+          level: info[0].level,
+          parent: parent,
+          children: children,
+          children_level: info[0].c_level,
+          bbox: formatBox(info[0].bbox)
+        };
+        res(reposnse);
```
Comment (dereklieu) [inline routes/admin.js]: typo that still works but should be fixed :P

File: routes/admin.js [hunk 8]
```diff
@@ -0,0 +1,250 @@
+'use strict';
+
+var Boom = require('boom');
+var knex = require('../connection');
+var getParents = require('../services/admin').getParents;
+var formatBox = require('../services/admin').formatBOX;
+Promise = require('bluebird');
```
Comment (dereklieu) [inline routes/admin.js]: I don't think you use bluebird at all here.

File: services/admin.js [hunk 9]
```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+/**
+ * parses then return all parent ids
+ * @param {Number} unitId numeric admin unit id
```
Comment (dereklieu) [inline services/admin.js]: This function looks like it expects a string, and should probably do a string conversion just to be safe anyway.

File: routes/admin.js [hunk 10]
```diff
@@ -0,0 +1,250 @@
...
+     * @apiDescription Returns information for admin unit with provided unit_id
+     * @apiVersion 0.1.0
+     *
+     * @apiParam {String} unit_id Admin unit id
+     *
+     * @apiSuccess {JSON} unitInfo object with unit info
+     * @apiSuccess {string} unitInfo.level admin unit level
+     * @apiSuccess {string} unitInfo.id unit id
+     * @apiSuccess {array} unitInfo.parent_ids unit's parent unit id(s)
+     * @apiSuccess {array} unitInfo.bbox unit's bounding box
```
Comment (dereklieu) [inline routes/admin.js]: This is short a few properties.

**Comment by maxgrossman** [comment]
@dereklieu - see updates per your review. thanks!

**Pull request merged**